cli/src/commands/exception.rs

```diff
@@ -0,0 +1,361 @@
+        Some(version) => version.to_string().into(),
+        None => prompt_version(&suggested_versions)?,
+    };
+    purl = purl.into_builder().with_version(version).build()?;
+
+    // Prompt for exception reason. if it wasn't supplied as an argument.
+    let reason = match reason {
+        Some(reason) => reason.into(),
+        None => prompt_reason()?,
+    };
+
+    // Build suppression API object.
+    let suppressions = vec![Suppression::Package(IgnoredPackage {
+        purl: Cow::Owned(purl.to_string()),
+        reason: Cow::Borrowed(&reason),
+    })];
+
+    match project {
+        Some(project_name) => {
+            let group = group.map(String::as_str);
+            let project_id = api.get_project_id(project_name, org, group).await?.to_string();
+            api.project_suppress(&project_id, &suppressions).await?;
+        },
+        None => api.group_suppress(org, group.unwrap(), &suppressions).await?,
+    }
+
+    print_user_success!("Successfully added suppression for {}", purl.to_string());
+
+    Ok(ExitCode::Ok)
+}
+
+/// Handle `phylum exception remove` subcommand.
+pub async fn handle_remove(api: &PhylumApi, matches: &ArgMatches, config: Config) -> CommandResult {
+    let ecosystem = matches.get_one::<String>("ecosystem");
+    let project = matches.get_one::<String>("project");
+    let version = matches.get_one::<String>("version");
+    let group = matches.get_one::<String>("group");
+    let name = matches.get_one::<String>("name");
+    let purl = matches.get_one::<String>("purl");
+    let tag = matches.get_one::<String>("tag");
+    let id = matches.get_one::<String>("id");
+    let org = config.org();
+
+    let mut exceptions = match matches.get_one::<String>("project") {
+        Some(project_name) => {
+            let group = group.map(String::as_str);
+            let project_id = api.get_project_id(project_name, org, group).await?.to_string();
+            api.project_preferences(&project_id).await?
+        },
+        None => api.group_preferences(config.org(), group.unwrap()).await?,
+    };
+
+    // Filter issue suppressions with CLI args.
+    if tag.is_some() || id.is_some() {
+        exceptions.ignored_issues.retain(|issue| {
+            id.is_none_or(|id| id == &issue.id) && tag.is_none_or(|tag| tag == &issue.tag)
+        });
+    }
+
+    // Filter package suppressions with CLI args.
+    if ecosystem.is_some() || name.is_some() || version.is_some() || purl.is_some() {
+        let purl = purl.map(|purl| Purl::from_str(purl));
+        let (ecosystem, namespace, name, version) = match purl {
+            Some(Ok(ref purl)) => {
+                let package_type = Cow::Owned(purl.package_type().to_string());
+                (Some(package_type), purl.namespace(), Some(purl.name()), purl.version())
+            },
+            Some(Err(err)) => return Err(err.into()),
+            None => (
+                ecosystem.map(Cow::Borrowed),
+                None,
```

Comment:
Should not matter since it's only a filter, not a hard requirement.

Reply:
It does matter. If `@angular/cli` is entered as the name, this code converts that into `let (namespace, name) = (None, Some("@angular/cli"));`. Because the namespace is incorrectly `None`, all packages are initially allowed through the filter, which is a frustrating bug on its own, but then because the name is incorrectly `Some("@angular/cli")`, no packages are allowed through the filter because that's not a valid PURL name for an NPM package. You probably want to replace namespace+name with the combined name before filtering.

Reply:
> all packages are initially allowed through the filter, which is a frustrating bug on its own

How is that a bug or frustrating? That's how filters work...

Reply:
If I specify that I want to remove a suppression for `@angular/cli` and the CLI asks me if i want to remove a suppression for `@storybook/cli`, that's obviously a bug to me.

Reply:
I wanted to avoid the additional complexity, but have added a dynamic resolver for namespace+name without known ecosystem. So it should now work more like what you expected I think?

Reply:
Wouldn't it be easier convert incoming PURLs into `Some(purl.package_type()), Some(purl.combined_name()), purl.version()` and filter on type+combined name+version? Then you'd never need to try interpreting an unknown combined name value as every possible package type.